File: l33tcode/minimum-swaps-to-make-sequences-increasing.go
package main

func minSwap(A []int, B []int) int {
	swapsRev, swapsSame := 1, 0

	for pos := 1; pos < len(A); pos++ {
		if A[pos-1] >= B[pos] || B[pos-1] >= A[pos] {
			swapsRev++
		} else if A[pos-1] >= A[pos] || B[pos-1] >= B[pos] {
			swapsRev, swapsSame = swapsSame+1, swapsRev
		} else {
			if swapsRev < swapsSame {
				swapsRev, swapsSame = swapsRev+1, swapsRev
			} else {
				swapsRev, swapsSame = swapsSame+1, swapsSame
			}
		}
	}

	result := 0

	if swapsRev < swapsSame {
		result = swapsRev
	} else {
		result = swapsSame
	}

	return result
}

File: l33tcode/minimum-swaps-to-make-sequences-increasing.py
class Solution:
    def minSwap(self, A: List[int], B: List[int]) -> int:
        swaps_rev, swaps_same = 1, 0

        for pos in range(1, len(A)):
            if A[pos - 1] >= B[pos] or B[pos - 1] >= A[pos]:
                swaps_rev += 1
            elif A[pos - 1] >= A[pos] or B[pos - 1] >= B[pos]:
                swaps_rev, swaps_same = swaps_same + 1, swaps_rev
            else:
                swaps_same = min(swaps_rev, swaps_same)
                swaps_rev = swaps_same + 1

        return min(swaps_rev, swaps_same)
